source/particle/generator/reference_cell.cc

```diff
@@ -0,0 +1,192 @@
+                    particle_positions.push_back(position_unit);
+                  }
+                else if (dim == 3)
+                  {
+                    for (unsigned int k = 0; k < number_of_particles[2]; ++k)
+                      {
+                        Point<dim> position_unit = Point<dim>(i * spacing[0] + spacing[0] / 2,
+                                                              j * spacing[1] + spacing[1] / 2,
+                                                              k * spacing[2] + spacing[2] / 2);
+                        particle_positions.push_back(position_unit);
+                      }
+                  }
+                else
+                  ExcNotImplemented();
+              }
+          }
+
+        return particle_positions;
+      }
+
+
+      template <int dim>
+      void
+      ReferenceCell<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            prm.enter_subsection("Generator");
+            {
+              prm.enter_subsection("Reference cell");
+              {
+                prm.declare_entry ("Number of tracers per cell per direction", "10",
+                                   Patterns::Double (0),
+                                   "Total number of tracers to create for each cell for each spatial dimension."
+                                   "The number is parsed as a floating point number (so that one can "
+                                   "specify, for example, '1e4' particles) but it is interpreted as "
+                                   "an integer, of course.");
+              }
+              prm.leave_subsection();
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+
+
+      template <int dim>
+      void
+      ReferenceCell<dim>::parse_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            prm.enter_subsection("Generator");
+            {
+              prm.enter_subsection("Reference cell");
+              {
+                for (int i=0; i<dim; i++)
+                  number_of_particles[i] = (unsigned int) (prm.get_double ("Number of tracers per cell per direction"));
+              }
+              prm.leave_subsection();
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+        prm.leave_subsection();
+      }
+    }
+  }
+}
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Generator
+    {
+      ASPECT_REGISTER_PARTICLE_GENERATOR(ReferenceCell,
+                                         "reference cell",
+                                         "Generates a uniform distribution of particles "
```

Comment:
I would add `per cell and spatial direction` to avoid the impression you have to provide `n_particles` locations